MyApplication/app/src/main/java/test/test/com/myapplication/MainActivity.java

```diff
@@ -1,42 +1,118 @@
 package test.test.com.myapplication;
 
+import android.location.Location;
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
 import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.FrameLayout;
 
-import test.test.com.myapplication.fragments.MainFragment;
+import com.sothree.slidinguppanel.SlidingUpPanelLayout;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import test.test.com.myapplication.fragments.SearchFragment;
+import test.test.com.myapplication.interfaces.ILocationCallBack;
+import test.test.com.myapplication.interfaces.OnRouteReady;
+import test.test.com.myapplication.interfaces.OnRouteReady1;
+import test.test.com.myapplication.utilities.LocationUtils;
+import test.test.com.myapplication.utilities.PermissionUtils;
+
+public class MainActivity extends AppCompatActivity implements OnRouteReady {
+    @Bind(R.id.sliding_layout)
+    SlidingUpPanelLayout mSlider;
+    @Bind(R.id.fragment_container)
+    FrameLayout mapLayout;
+    private android.app.Fragment mainFragment;
```

Comment:
This is wrong as you are also using the fragment from the support library - you should check that there are no collisions like that - it will cause devices to crash...